jwk/handler.go

```diff
@@ -196,24 +197,12 @@ func (h *Handler) Create(w http.ResponseWriter, r *http.Request, ps httprouter.P
 		h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
 	}
 
-	generator, found := h.r.KeyGenerators()[keyRequest.Algorithm]
-	if !found {
-		h.r.Writer().WriteErrorCode(w, r, http.StatusBadRequest, errors.Errorf("Generator %s unknown", keyRequest.Algorithm))
-		return
-	}
-
-	keys, err := generator.Generate(keyRequest.KeyID, keyRequest.Use)
-	if err != nil {
+	if keys, err := h.r.KeyManager().GenerateKeySet(r.Context(), set, keyRequest.KeyID, keyRequest.Algorithm, keyRequest.Use); err == nil {
+		keys = ExcludeOpaquePrivateKeys(keys)
```

Comment:
What would happen if these were included?

Reply:
You would get "json: error calling MarshalJSON for type jose.JSONWebKey: square/go-jose: unknown key type '*cryptosigner.cryptoSigner'"
Private key is opaque and would not contain any info about private key so does it make sense to return only alg, use, key values? Or do you have idea how to indicate that it is opaque key when not using this filter?

Reply:
That's good! :)